python/src/netlist.py

```diff
@@ -126,24 +146,38 @@ def persist(self, file: File) -> None:
             if len(device_line) == 0 or not device_line[0].isalpha():
                 continue
 
-            old_width_str = re.search(DEVICE_WIDTH_REGEX, device_line, re.IGNORECASE).group()
-            old_length_str = re.search(DEVICE_LENGTH_REGEX, device_line, re.IGNORECASE).group()
-            old_fingers_str = re.search(DEVICE_FINGERS_REGEX, device_line, re.IGNORECASE).group()
+            match = re.search(DEVICE_WIDTH_REGEX, device_line, re.IGNORECASE)
+            if match:
+                old_width_str = match.group()
+            else:
+                raise ValueError(
+                    f"No match of {DEVICE_WIDTH_REGEX} found in {device_line}"
+                )
+
+            match = re.search(DEVICE_LENGTH_REGEX, device_line, re.IGNORECASE)
+            if match:
+                old_length_str = match.group()
+            else:
+                raise ValueError(
+                    f"No match of {DEVICE_LENGTH_REGEX} found in {device_line}"
+                )
+
+            match = re.search(DEVICE_FINGERS_REGEX, device_line, re.IGNORECASE)
+            if match:
+                old_fingers_str = match.group()
+            else:
+                raise ValueError(
+                    f"No match of {DEVICE_FINGERS_REGEX} found in {device_line}"
+                )
```

Comment:
Should be able to make another function like
```python3
def _match(line: str, regex: str) -> str:
  match = re.search(regex, line, re.IGNORECASE)
  if not match:
    raise ValueError(f"No match of {regex} found in {line}")
  return match.group()
```

Reply:
I will leave this for another PR